web/src/features/map/Map.tsx

```diff
@@ -153,14 +156,46 @@ export default function MapPage({ onMapLoad }: MapPageProps): ReactElement {
 
   useEffect(() => {
     // Run on first load to center the map on the user's location
-    if (!map || isError || !isFirstLoad) {
+    if (!map || isError || !isFirstLoad || !isSourceLoaded || !callerLocation) {
       return;
     }
-    if (data?.callerLocation && !selectedZoneId) {
-      map.flyTo({ center: [data.callerLocation[0], data.callerLocation[1]] });
+    if (callerLocation && !selectedZoneId) {
+      map.flyTo({ center: [callerLocation[0], callerLocation[1]] });
+
+      const handleIdle = () => {
+        if (map.isSourceLoaded(ZONE_SOURCE) && map.areTilesLoaded() && callerLocation) {
```

Comment:
`callerLocation` should always be available at this stage
```suggestion
        if (map.isSourceLoaded(ZONE_SOURCE) && map.areTilesLoaded()) {
```